mtcpclient/groupofconnections.go

```diff
@@ -2,15 +2,34 @@ package mtcpclient
 
 import (
 	"fmt"
+	"strconv"
+	"time"
 
 	"github.com/dachad/tcpgoon/tcpclient"
 )
 
-type GroupOfConnections []tcpclient.Connection
+// GroupOfConnections aggregates all the running connections plus some general metrics
+type GroupOfConnections struct {
+	connections []tcpclient.Connection
+	metrics     gcMetrics
+}
+
+type gcMetrics struct {
+	maxConcurrentEstablished int
+}
+
+func newGroupOfConnections(numberConnections int) *GroupOfConnections {
+	gc := new(GroupOfConnections)
+	gc.connections = make([]tcpclient.Connection, numberConnections)
+	gc.metrics = gcMetrics{
```

Comment:
why do you initialise some "instances" using the new() function, but not here?

Reply:
gc is a pointer, gc.metrics no